sprockets_end_to_end_test.py

```diff
@@ -0,0 +1,219 @@
+"""End-to-end tests for all of Sprockets."""
+
+import os
+import shutil
+import tempfile
+import unittest
+
+import test_driver
+
+# A default manifest to use in the tests below. If used, make sure
+# the STL file under test uses the filename "sprockets_end_to_end.stl" to match
+# what's in the manifest here.
+_DEFAULT_MANIFEST_FILENAME = 'sprockets_end_to_end.test'
+_DEFAULT_MANIFEST_CONTENTS = (
+"""
+{
+  'stl_files': [
+    'sprockets_end_to_end.stl',
+  ],
+
+  'roles': [  # Role information
+     { 'role': 'example::rReceiver',
+       'ipAddress': '0.0.0.0',
+     },
+  ],
+
+  'constants': {
+    'example::kHelloWorld': 'Hello, world!',
+  },
+
+  'test': ['example::rReceiver'],
+}
+""")
```

Comment:
I thought it might make it easier to read the test to be able to see the file contents with the error in-line. But I can pull them out into files if that's more clear